src/plugins/StatusBar/StatusBar.js

```diff
@@ -106,24 +106,58 @@ module.exports = (props) => {
   }
 
   const width = typeof progressValue === 'number' ? progressValue : 100
+  const isHidden = uploadState === STATE_WAITING && !props.newFiles > 0
 
   return html`
     <div class="UppyStatusBar is-${uploadState}"
-                aria-hidden="${uploadState === STATE_WAITING}"
-                title="">
-      <progress style="display: none;" min="0" max="100" value=${progressValue}></progress>
+         aria-hidden="${isHidden}">
       <div class="UppyStatusBar-progress ${progressMode ? `is-${progressMode}` : ''}"
-           style="width: ${width}%"></div>
+           style="width: ${width}%"
+           role="progressbar"
+           aria-valuemin="0"
+           aria-valuemax="100"
+           aria-valuenow="${progressValue}"></div>
       ${progressBarContent}
+      <div class="UppyStatusBar-actions">
+        ${props.newFiles ? UploadBtn(props) : ''}
+        ${props.error ? RetryBtn(props) : ''}
+      </div>
     </div>
   `
 }
 
+const UploadBtn = (props) => {
+  return html`<button type="button"
+                      class="UppyStatusBar-actionBtn UppyStatusBar-actionBtn--upload"
+                      aria-label="${props.i18n('uploadXFiles', { smart_count: props.newFiles })}"
+                      onclick=${props.startUpload}>
+                ${props.inProgress
+                  ? props.i18n('uploadXNewFiles', { smart_count: props.newFiles })
+                  : props.i18n('uploadXFiles', { smart_count: props.newFiles })
+                }
+              </button>`
+}
+
+const RetryBtn = (props) => {
+  return html`<button type="button"
+                      class="UppyStatusBar-actionBtn UppyStatusBar-actionBtn--retry"
+                      aria-label="${props.i18n('retryUpload')}"
+                      onclick=${props.retryAll}>
+                ${props.i18n('retry')}
+              </button>`
+}
```

Comment:
I wonder if we could pass these actions in as an option somehow? for a standalone StatusBar, not inside the Dashboard, having the upload button is a bit strange.

maybe so that the dashboard can do something like

```js
this.core.use(StatusBar, {
  id: `${this.id}-StatusBar`,
  actions (state) {
    return [ hasNewFiles && UploadBtn({ /* xyz */ }), hasError && RetryBtn({ /* xyz */ }) ]
  }
})
```

Reply:
or more simply, we could also have a boolean option that toggles the visibility of the action buttons.

Reply:
I’ve added an option to hide the Upload button, do you think similar is needed for each of the other buttons or all of them?

> for a standalone StatusBar, not inside the Dashboard, having the upload button is a bit strange

Is is though? You have a DragDrop area, stick StatusBar underneath, and now you have a functioning uploader! :-)